### ISSUE TITLE
Blank dashboard tab w/ flower 0.8.2

### ISSUE THREAD

**Issue by dwiest** [mher/flower#397]
Flower 0.7.3 was working okay for me and showing data on all tabs.  After upgrading to 0.8.2 most of the data is missing with the exception of the Broker tab which does show the correct queues, idle times, etc.

![screen shot 2015-05-19 at 5 09 26 pm](https://cloud.githubusercontent.com/assets/12519645/7715062/dc96c232-fe49-11e4-88e9-b1ec3ac25416.png)

![screen shot 2015-05-19 at 5 11 01 pm](https://cloud.githubusercontent.com/assets/12519645/7715081/23d7eed2-fe4a-11e4-93ea-7910e37d5fc1.png)

Some debug output is listed below.  The broker_api URL looks odd, but is correct; I've wrapped RabbitMQ's MochiWeb interface with stunnel so we can use HTTPS with client certificates.  On the host with flower I'm running stunnel in client mode to unwrap the TLS and allow connections to localhost to communicate using HTTP.

If I downgrade to 0.7.3 and run the same command listed below, then I do see all of the expected data.  

[root@master-compute6b ~]# /opt/app/.virtualenvs/app/bin/python /opt/app/.virtualenvs/app/bin/flower --broker=amqp://master-compute6:5672 --broker_api=http://localhost:15672/api/ --address=127.0.0.1 --debug
[I 150519 17:02:41 command:114] Visit me at http://127.0.0.1:5555
[I 150519 17:02:41 command:116] Broker: amqp://guest@master-compute6:5672//
[I 150519 17:02:41 command:119] Registered tasks: 
    ['celery.backend_cleanup',
     'celery.chain',
     'celery.chord',
     'celery.chord_unlock',
     'celery.chunks',
     'celery.group',
     'celery.map',
     'celery.starmap']
[D 150519 17:02:41 command:121] Settings: {'cookie_secret': 'Mw/2YLuHRpmEiDIqkQ1TcBoxM3t5e0mEtpU8nlFPdKo=',
     'debug': True,
     'login_url': '/login',
     'static_path': '/opt/app/.virtualenvs/app/lib/python2.7/site-packages/flower/static',
     'template_path': '/opt/app/.virtualenvs/app/lib/python2.7/site-packages/flower/templates'}
[D 150519 17:02:41 control:28] Updating all worker's cache...
[I 150519 17:02:41 mixins:225] Connected to amqp://guest@master-compute6:5672//
[D 150519 17:02:43 app:49] Successfully updated worker cache
[I 150519 17:03:27 web:1825] 200 GET /dashboard (127.0.0.1) 27.65ms
[I 150519 17:03:27 web:1825] 304 GET /static/css/bootstrap.css?v=7b4cd90ae3b1616f1cf0b93bf5a7769b (127.0.0.1) 13.88ms
[I 150519 17:03:28 web:1825] 304 GET /static/css/bootstrap-responsive.css?v=702e8485242b3ae5b4ce75a5ede13acb (127.0.0.1) 1.58ms
[I 150519 17:03:28 web:1825] 304 GET /static/css/jquery-ui.css?v=11ba5be990454fbbc957fcdf55e339ca (127.0.0.1) 1.01ms
[I 150519 17:03:28 web:1825] 304 GET /static/css/bootstrap-datetimepicker.min.css?v=7468d8a6368bc2a64ad76bce62b9beb2 (127.0.0.1) 1.67ms
[I 150519 17:03:28 web:1825] 304 GET /static/css/flower.css?v=e58378e50914b035bc25ae7b3a160b48 (127.0.0.1) 1.60ms
[I 150519 17:03:28 web:1825] 304 GET /static/css/rickshaw.min.css?v=48a108292e153ef3cfb53c32283b3d2c (127.0.0.1) 1.55ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/jquery-1.7.2.min.js?v=b8d64d0bc142b3f670cc0611b0aebcae (127.0.0.1) 1.92ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/jquery-ui-1-8-15.min.js?v=88d9f574687d11e3ee3c36b97ac37ffc (127.0.0.1) 2.54ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-transition.js?v=871f492dffbee74e6a7134159ac6022b (127.0.0.1) 1.56ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-alert.js?v=12586670237d66f7ddc0ba3c6565faff (127.0.0.1) 1.56ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-modal.js?v=5fe4c14f9cbba0a03dfd2cf0a74bb812 (127.0.0.1) 1.51ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-dropdown.js?v=f0a761d953362eb3c150f6ac876a4638 (127.0.0.1) 1.47ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-scrollspy.js?v=f7f5435ab136c477b98c1cff0a09c749 (127.0.0.1) 1.73ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-tab.js?v=ca0b95948cc93f4ec18cc3013726a220 (127.0.0.1) 1.43ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-tooltip.js?v=cbba0d4d7ed3f007b8d287f1b5eee7e8 (127.0.0.1) 1.62ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-popover.js?v=69df927a7b524b87ca3badade4fa4e09 (127.0.0.1) 1.74ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-button.js?v=8b493affaa8e27831d3162d46807b624 (127.0.0.1) 1.66ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-collapse.js?v=e8ddac0b5dd49cfbcf7d3ca8b0098d7b (127.0.0.1) 1.55ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-carousel.js?v=fc8cbc40f39316b8b567b3b96efe9044 (127.0.0.1) 1.56ms
[I 150519 17:03:28 web:1825] 304 GET /static/js/bootstrap-typeahead.js?v=a71319e43efd22bf29161d0e75b892b7 (127.0.0.1) 1.57ms
[I 150519 17:03:29 web:1825] 304 GET /static/js/d3.min.js?v=eb68d3d1035789d336b285373046b550 (127.0.0.1) 1.77ms
[I 150519 17:03:29 web:1825] 304 GET /static/js/d3.layout.min.js?v=4d73dea16077b0d7d128ecf7a4c20752 (127.0.0.1) 1.59ms
[I 150519 17:03:29 web:1825] 304 GET /static/js/rickshaw.min.js?v=fc927b6dd64118caa563b711bcb2f130 (127.0.0.1) 1.93ms
[I 150519 17:03:29 web:1825] 304 GET /static/js/bootstrap-datetimepicker.min.js?v=8880b6a34ee02b5cb6a75f92b3a7ddc9 (127.0.0.1) 0.88ms
[I 150519 17:03:29 web:1825] 304 GET /static/js/flower.js?v=265f42c20347dea9f7038ae3d9b354bb (127.0.0.1) 0.79ms
[I 150519 17:03:29 web:1825] 200 GET /static/favicon.ico?v=fcb9ede4815b4949f75cbead02450a72 (127.0.0.1) 1.87ms
[D 150519 17:03:29 dashboard:69] Starting a timer for dashboard updates

**Comment by mher** [comment]
Can you try the master version?

**Comment by dwiest** [comment]
I'm seeing the same issue when testing against the master version. 

In the RabbitMQ server's log I can see the AMQP connections being accepted.

=INFO REPORT==== 7-Aug-2015::15:19:23 ===
accepting AMQP connection <0.17533.17> (192.168.100.189:35822 -> 192.168.112.99:5672)

Here is the debug output that is generated when viewing the landing (Dashboard) page.  I'm testing on a different host than before, so the broker URLs are a bit different.  

(mypy)[dwiest@716161-compute2 site-packages]$ python flower --broker=amqp://master-compute1:5672 --broker_api=http://localhost:15672/api/  --address=192.168.100.189 --debug=true
[I 150807 14:45:42 command:113] Visit me at http://192.168.100.189:5555
[I 150807 14:45:42 command:115] Broker: amqp://guest:**@master-compute1:5672//
[I 150807 14:45:42 command:118] Registered tasks: 
    ['celery.backend_cleanup',
     'celery.chain',
     'celery.chord',
     'celery.chord_unlock',
     'celery.chunks',
     'celery.group',
     'celery.map',
     'celery.starmap']
[D 150807 14:45:42 command:120] Settings: {'cookie_secret': 'eSR13d5wR0G0sASvNKnhogpFmsBRUEPPhbfMRgMUbeE=',
     'debug': True,
     'login_url': '/login',
     'static_path': '/home/dwiest/mypy/lib/python2.6/site-packages/flower/static',
     'template_path': '/home/dwiest/mypy/lib/python2.6/site-packages/flower/templates'}
[D 150807 14:45:42 control:29] Updating all worker's cache...
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 channel:438] Channel open
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:754] Start from server, version: 0.9, properties: {u'information': u'Licensed under the MPL.  See http://www.rabbitmq.com/', u'product': u'RabbitMQ', u'copyright': u'Copyright (C) 2007-2014 GoPivotal, Inc.', u'capabilities': {u'exchange_exchange_bindings': True, u'connection.blocked': True, u'authentication_failure_close': True, u'basic.nack': True, u'per_consumer_qos': True, u'consumer_priorities': True, u'consumer_cancel_notify': True, u'publisher_confirms': True}, u'cluster_name': u'rabbit@master-compute1', u'platform': u'Erlang/OTP', u'version': u'3.3.3'}, mechanisms: [u'PLAIN', u'AMQPLAIN'], locales: [u'en_US']
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 connection:640] Open OK!
[D 150807 14:45:42 channel:438] Channel open
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 channel:438] Channel open
[D 150807 14:45:42 channel:438] Channel open
[I 150807 14:45:42 mixins:225] Connected to amqp://guest:**@master-compute1:5672//
[D 150807 14:45:42 channel:438] Channel open
[D 150807 14:45:42 channel:80] using channel_id: 1
[D 150807 14:45:42 channel:438] Channel open
[D 150807 14:45:45 app:49] Successfully updated worker cache
[I 150807 14:45:47 web:1908] 304 GET /dashboard (192.168.237.167) 19.47ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/bootstrap.css?v=7b4cd90ae3b1616f1cf0b93bf5a7769b (192.168.237.167) 5.92ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/flower.css?v=c956fd18aac1e617570f5366d11ad4fd (192.168.237.167) 0.63ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/bootstrap-responsive.css?v=702e8485242b3ae5b4ce75a5ede13acb (192.168.237.167) 0.66ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/rickshaw.min.css?v=48a108292e153ef3cfb53c32283b3d2c (192.168.237.167) 0.71ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/bootstrap-datetimepicker.min.css?v=7468d8a6368bc2a64ad76bce62b9beb2 (192.168.237.167) 0.70ms
[I 150807 14:45:47 web:1908] 304 GET /static/css/jquery-ui.css?v=11ba5be990454fbbc957fcdf55e339ca (192.168.237.167) 0.79ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/jquery-1.7.2.min.js?v=b8d64d0bc142b3f670cc0611b0aebcae (192.168.237.167) 0.89ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/jquery-ui-1-8-15.min.js?v=88d9f574687d11e3ee3c36b97ac37ffc (192.168.237.167) 1.12ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-transition.js?v=871f492dffbee74e6a7134159ac6022b (192.168.237.167) 0.63ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-alert.js?v=12586670237d66f7ddc0ba3c6565faff (192.168.237.167) 0.62ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-modal.js?v=5fe4c14f9cbba0a03dfd2cf0a74bb812 (192.168.237.167) 0.67ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-dropdown.js?v=f0a761d953362eb3c150f6ac876a4638 (192.168.237.167) 0.66ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-scrollspy.js?v=f7f5435ab136c477b98c1cff0a09c749 (192.168.237.167) 0.62ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-tab.js?v=ca0b95948cc93f4ec18cc3013726a220 (192.168.237.167) 0.66ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-tooltip.js?v=cbba0d4d7ed3f007b8d287f1b5eee7e8 (192.168.237.167) 0.64ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-popover.js?v=69df927a7b524b87ca3badade4fa4e09 (192.168.237.167) 0.60ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-button.js?v=8b493affaa8e27831d3162d46807b624 (192.168.237.167) 0.62ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-collapse.js?v=e8ddac0b5dd49cfbcf7d3ca8b0098d7b (192.168.237.167) 0.64ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-carousel.js?v=fc8cbc40f39316b8b567b3b96efe9044 (192.168.237.167) 0.61ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/d3.min.js?v=eb68d3d1035789d336b285373046b550 (192.168.237.167) 0.75ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-typeahead.js?v=a71319e43efd22bf29161d0e75b892b7 (192.168.237.167) 0.64ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/d3.layout.min.js?v=4d73dea16077b0d7d128ecf7a4c20752 (192.168.237.167) 0.64ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/rickshaw.min.js?v=fc927b6dd64118caa563b711bcb2f130 (192.168.237.167) 0.78ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/bootstrap-datetimepicker.min.js?v=8880b6a34ee02b5cb6a75f92b3a7ddc9 (192.168.237.167) 0.71ms
[I 150807 14:45:47 web:1908] 304 GET /static/js/flower.js?v=c8dc7f7da7155241f49556fe1d5640df (192.168.237.167) 0.75ms
[I 150807 14:45:47 web:1908] 200 GET /static/favicon.ico?v=fcb9ede4815b4949f75cbead02450a72 (192.168.237.167) 0.78ms
[D 150807 14:45:47 dashboard:71] Starting a timer for dashboard updates

**Comment by dwiest** [comment]
Apologies, it appears that I just wasn't patient enough.  I walked away from my computer for about ten minutes and when I came back the worker data had populated the Dashboard screen.

I'll try testing against 0.8.3 again before closing this issue.

**Comment by dwiest** [comment]
I have tested against 0.7.3, 0.8.3 and 0.9 and there does appear to be a problem retrieving worker data.

0.7.3 -- data for 76 workers was returned in less than 10 seconds.
0.8.3 -- no worker data showed up after 14 minutes and only 6 of 76 workers showed up after 20 minutes.
0.9.0 -- no worker data after 4 minutes, 44/76 workers showed up after 15 minutes.  Sometimes workers only appear after I restart them.

With versions 0.8.3 and 0.9.0, If I restart workers they show up right away as Online, but their status then changes to Offline and stays that way. 

Workers are using celery version 3.1.6 and the RabbitMQ server is version 3.3.3.

**Comment by mher** [comment]
Do you see all servers when you do `celery inspect ping`?

**Comment by mher** [comment]
Have you tried to update the dashboard? There is a refresh button on the dashboard page.

**Comment by dwiest** [comment]
Hi Mher, I appreciate you taking the time to reply.

I'm unable to reproduce the problems I was having last week; flower seems to be working perfectly. 

We did have some issues on Friday with certain tasks causing celery workers to use up all of system memory and trigger the OOM killer.  The host I was using for testing was affected and ended up being rebooted later in the day after our code was corrected.  That's my best guess as to what's different this week.

I'm going to go ahead and deploy the master version of flower to our environments.  I'll close this issue once I've confirmed they look okay.

BTW, the new version of flower looks great!

**Comment by mher** [comment]
@dwiest You can use [--max_tasks](https://flower.readthedocs.org/en/latest/config.html#max-tasks) option to limit the number of in-memeory tasks.

**Comment by dwiest** [comment]
I finally figured out what was going on.

I run RabbitMQ's mochiweb interface on localhost and then use stunnel to provide access via HTTPS to remote users where I also require they present a client certificate.

On remote hosts running flower I use stunnel to connect to mochiweb over HTTPS and then provide an HTTP service on localhost for flower to use.  I have flower bind to localhost and once again use stunnel to provide access via HTTPS to remote users with a client certificate.

The version of stunnel I'm running is a few releases behind and it was misbehaving when I would restart flower.  Sometimes stunnel would exit, but other times it would just not respond properly.  This resulted in behavior like workers initially showing up as offline when they were really online, or workers reporting in very slowly.  I found that if I restarted stunnel when I was experiencing this behavior, then the workers showed up immediately with the proper status.

I know this setup is complicated, but I like to use multiple layers of access controls for these types of services.
